### ISSUE TITLE
Antivirus false alarms: can we fight them?

### ISSUE THREAD

**Issue by d0vgan** [winsiderss/systeminformer#1715]
### Description of the feature, modification, idea or suggestion

For each version of SystemInformer, even though they are signed, I regularly get antivirus false positives for the following files:
- peview.exe
- plugins\ExtendedServices.dll
- plugins\HardwareDevices.dll
- plugins\NetworkTools.dll
- plugins\ToolStatus.dll
- plugins\Updater.dll
- plugins\UserNotes.dll
- plugins\WindowExplorer.dll

Actually, I have the very same situation with my own open-source project written in pure C (false positives from several antiviruses for the .dll file built by different versions of Visual Studio):
https://github.com/d0vgan/AkelPad-Plugs-QSearch/releases/tag/v79

Looks like antiviruses do not like pure C code in general. But shouldn't this be investigated and fixed by antivirus developers instead of creating problems and damaging the reputation of software products affected by false positives?
What mechanisms do we have to fight this situation? Moreover, as SystemInformer's binaries are signed, why this situation happens at all?
Here you can find an example of idiotic antiviruses with false positives for "peview.exe". They are 25 from 69, damn it all:
https://www.virustotal.com/gui/file/6de2a8b604f2b512cf2630e5e5c0fe9132a2b07b85a0ba82deff5a9e973094eb/detection/f-6de2a8b604f2b512cf2630e5e5c0fe9132a2b07b85a0ba82deff5a9e973094eb-1683266885

**Comment by Masamune3210** [comment]
Trust me, it's a very well known issue. Unfortunately, the only ones that can stop it from happening are the ones that started it. All users and devs can do is submit false positive reports and hope that it gets fixed

**Comment by Shadetail** [comment]
Yeah, same issue here. System Informer just suddenly disappeared from my system. And upon trying to reinstall I got blocked by virus warning, chrome won't even allow me to download it. 
![image](https://user-images.githubusercontent.com/11357447/236509620-7c3c6890-a0fb-4dd7-bf4e-ea4d7b834ba0.png)

**Comment by BinToss** [comment]
I'd like to note that binaries of nightly release 3.0.6550 were flagged as `Trojan:Win32/Malgent` likely because SystemInformer.exe and systeminformer-setup.exe were **NOT** signed. This setup file was downloaded by the self-update mechanism, if that matters.
Until this update, System Informer nightly releases hadn't been flagged on my PCs since upgrading from the final Process Hacker nightly release.
![image](https://user-images.githubusercontent.com/7243190/236554878-ab4df5ee-131c-4f5f-8a25-937a59fa6dfb.png)

**Comment by KyleKolander** [comment]
In case it helps anyone, I couldn't uninstall until I restored the quarantined items.  Then I downloaded the previous version and installed that.  Defender did not flag it as a virus or quarantine any of it.

https://github.com/winsiderss/si-builds/releases/tag/3.0.6548

**Comment by methuzla** [comment]
System Informer stopped working,
Defender had killed it.

All manner of problems trying to reinstall.

https://www.systeminformer.com/nightly.php
 tried to download 'setup'.

Each attempt to download installer met with "Smart" screen detected a problem, calls it a trojan, blocked, deleted.

Registered systeminformer as a Defender exception (in several ways), and changed properties to accept external app, Say 'keep', two places.
No joy. Just deletes it. Arrogance.

Suspended "Smart" screen & Defender, tried again. Still it deletes it, "Trojan" will damage your machine. Gone.
(!)

Defender "exception" settings are not honoured, even after reboot.
Defender ignores me and is certain it knows better. Very poor implementation.

Incidentally, virustotal.com has score zero, no problems.

Eventually achieved re-install by running installer found inside program-files folder, which bypassed downloads.
I suppose new potential users cannot succeed.

Presumably part of the MS monopoly attempt, plus incompetence.

**Comment by revelator** [comment]
norton especially is a huge pita with this tool grrr.

**Comment by d0vgan** [comment]
I had successfully contacted Microsoft Defender, ESET, VBA32, Fortinet, Sophos, Avira, Symantec and Ikarus by e-mail regarding false positives in QSearch.dll mentioned above. Those false positives have been removed. Probably the same could be done for System Informer by its authors (developers).
I had also contacted McAfee, and there were a lot of e-mail iterations where they asked me for more and more details (including my full name, address, phone number and so on). Finally they suggested me to send the files under the question as the reply to their e-mail (even though these files are publically available from github), but these files could not be delivered because of e-mail  security policies. This is where everything has stopped with McAfee, even though I wrote them about the delivery problem and suggested to download the files from github by direct links (they did not reply anything to that).
Finally, such antiviruses as Elastic, Zillya, Alibaba, Antiy, Cylance, MaxSecure, Palo Alto, Panda Security, Tencent and TrendMicro have not replied by e-mail at all, even though I mentioned that Microsoft Defender, ESET, VBA32, Fortinet, Sophos, Avira, Symantec and Ikarus had already confirmed the false positives. To my opinion, this absense of response from these teams illustrates the level of professionalism, quality and reliability of their antiviruses.

**Comment by connorjak** [comment]
Before you do this, do a sanity check to ensure you're not about to reactivate an actual trojan.

"What you're supposed to do" resolution steps:
1. Open Windows Security
2. Virus & threat protection -> Protection history
3. Open the latest event (probably Threat Quarantined)
4. Actions -> Restore

**Comment by terenzif** [comment]
Why don't you publish an ufficial signed release? The last one is dated Mar 29, 2016!!

**Comment by KyleKolander** [comment]
> In case it helps anyone, I couldn't uninstall until I restored the quarantined items. Then I downloaded the previous version and installed that. Defender did not flag it as a virus or quarantine any of it.
> 
> https://github.com/winsiderss/si-builds/releases/tag/3.0.6548

Scratch that.  I restarted my machine and Defender now quarantined and blocked Systeminformer.exe due to Trojan:Script/Conteban.A!ml.  This feels bad.  I'm going to uninstall it.  I'll keep checking in from time to time, but I can't justify this risk.  Really unfortunate because it's such a useful tool.

**Comment by KyleKolander** [comment]
> Why don't you publish an ufficial signed release? The last one is dated Mar 29, 2016!!

This.

**Comment by Masamune3210** [comment]
> > In case it helps anyone, I couldn't uninstall until I restored the quarantined items. Then I downloaded the previous version and installed that. Defender did not flag it as a virus or quarantine any of it.
> > https://github.com/winsiderss/si-builds/releases/tag/3.0.6548
> 
> Scratch that. I restarted my machine and Defender now quarantined and blocked Systeminformer.exe due to Trojan:Script/Conteban.A!ml. This feels bad. I'm going to uninstall it. I'll keep checking in from time to time, but I can't justify this risk. Really unfortunate because it's such a useful tool.

Risk of what, exactly?  SI isn't malicious, and Defender is telling you that it's best guess (which is usually wrong because heuristic-based machine learning has some pretty garbage tier issues that still haven't been worked out) is that it's malicious, when in reality it's just because it has the audacity to do things Defender thinks a normal program shouldn't be trying to do.

**Comment by KyleKolander** [comment]
> Is now already totally clean on ESET:
> https://forum.eset.com/topic/36280-false-alerts-of-system-informer/

Uh, I read through the entire thread from your link and there are only two random people (you and one other person) pleading to get ESET to stop flagging SI, while completely discounting (or outright ignoring) the evidence and arguments presented by others.

My take away was that Process Explorer doesn't get flagged because it is WHQL certified.  SI is not.  This is not my area of expertise, though.

As an aside, I don't know you or your background, but I'll offer a little constructive feedback - take it for what you will.

Bringing up (multiple times) things like whether or not the world is flat is going to result in most people not taking your point seriously. 

Also, you're missing the point that others are trying to make.  No one is arguing with you about whether ESET is flagging SI.  The point is that, whether through personal choice or due to their employer's choice, many people use other AV.  While others are saying the grass is green, you keep saying that the sky is blue.  If you take a moment to step back and look at from others' perspective, you'll find that the issue is not as cut and dry as you've been stating.

Don't shoot the messenger.  I'm honestly trying to help.

**Comment by shawnholt** [comment]
This should be the top issue. Nobody is going to try this out until resolved. https://www.virustotal.com/gui/file/8ee9d84de50803545937a63c686822388a3338497cddb660d5d69cf68b68f287?nocache=1

**Comment by SirWobbyTheFirst** [comment]
> > > Is now already totally clean on ESET:
> > > https://forum.eset.com/topic/36280-false-alerts-of-system-informer/
> > 
> > 
> > Uh, I read through the entire thread from your link and there are only two random people (you and one other person) pleading to get ESET to stop flagging SI, while completely discounting (or outright ignoring) the evidence and arguments presented by others.
> > My take away was that Process Explorer doesn't get flagged because it is WHQL certified. SI is not. This is not my area of expertise, though.
> > As an aside, I don't know you or your background, but I'll offer a little constructive feedback - take it for what you will.
> > Bringing up (multiple times) things like whether or not the world is flat is going to result in most people not taking your point seriously.
> > Also, you're missing the point that others are trying to make. No one is arguing with you about whether ESET is flagging SI. The point is that, whether through personal choice or due to their employer's choice, many people use other AV. While others are saying the grass is green, you keep saying that the sky is blue. If you take a moment to step back and look at from others' perspective, you'll find that the issue is not as cut and dry as you've been stating.
> > Don't shoot the messenger. I'm honestly trying to help.
> 
> 1. The ESET guys simply checked the code and then decided, that really good AVs always should when somebody open it as case. But sadly many bad ones are 99% only based of public databases and never really check at all, that's the easiest way to make money.
> 2. And Process Explorer is owned of MS SysInternals and luckily they not started yet a war inside the company. LOL

You're still not paying attention to him. Re-read it again and slowly this time.

**Comment by SirWobbyTheFirst** [comment]
> This should be the top issue. Nobody is going to try this out until resolved. https://www.virustotal.com/gui/file/8ee9d84de50803545937a63c686822388a3338497cddb660d5d69cf68b68f287?nocache=1

This right here is proof that it isn't just Defender that is flagging it. Hell on that link, Defender doesn't even flag it as malware. ESET has read through the code and whitelisted it, fantastic, do you want a medal?

**Comment by SirWobbyTheFirst** [comment]
> > > This should be the top issue. Nobody is going to try this out until resolved. https://www.virustotal.com/gui/file/8ee9d84de50803545937a63c686822388a3338497cddb660d5d69cf68b68f287?nocache=1
> > 
> > 
> > This right here is proof that it isn't just Defender that is flagging it. Hell on that link, Defender doesn't even flag it as malware. ESET has read through the code and whitelisted it, fantastic, do you want a medal?
> 
> Enjoy your (useless) databases and limited mind. Not my problem at all...

Cry harder 5nose.

**Comment by Masamune3210** [comment]
I don't know where the idea that most AV solutions use public databases comes from, but it's dumb. Sharing access to a database entry for something between vendors isn't a public database, is actually smart, and AV vendors fiercely defend their own databases and engines more often than not.

**Comment by dmex** [comment]
> shouldn't this be investigated and fixed by antivirus developers instead of creating problems and damaging the reputation of software products affected by false positives?

Yes.

> What mechanisms do we have to fight this situation?

There are no exemptions for antivirus software from business and consumer laws. I would suggest you speak with a lawyer about unconscionable conduct, false or misleading claims, defamation or damages. These are legal requirements the industry is required to follow and it's a mandatory requirement for businesses to prove any claims or statements especially when they're claiming something is malicious.

> using the corresponding web page provided by the antivirus maintainers). Those false positives have been removed. Probably the same could be done for System Informer

There's ongoing discussions and other vendors haven't completed their analysis yet. 

> here you can find an example of idiotic antiviruses with false positives for "peview.exe". 

peviewer literally doesn't do anything so I don't understand why they're labeling it malicious. The other funny thing is if you download the bin.zip and upload the \arm64\systeminformer.exe there's 0 detections but the \amd64\systeminformer.exe binary has 28 detections... It's the same binary with the exact same code and compiled at the same time on the same machine but somehow has completely different results 🤷 

> Agreed, and kudos to ESET for pulling their finger out

ESET is the first vendor who's provided information to date. I'm impressed with how honest they've been with customers and confirmed there are no security issues after reviewing our sources 👍

> This should be the top issue. Nobody is going to try this out until resolved. 

We already have an existing issue so I'm closing this as a duplicate of #1668

**Comment by dmex** [comment]
Duplicate of #1668
